rebasebot/bot.py

```diff
@@ -76,7 +76,13 @@ def _commit_go_mod_updates(gitwd: git.Repo, source: GitHubBranch) -> None:
                 full_path = os.path.join(module_base_path, filename)
                 if not os.path.exists(full_path):
                     continue
-                gitwd.remotes.source.repo.git.checkout(f"source/{source.branch}", full_path)
+                try:
+                    gitwd.remotes.source.repo.git.checkout(f"source/{source.branch}", full_path)
+                except Exception as err:
+                    if "did not match any file(s) known to git" in str(err):
```

Comment:
Ideally, you'd look at the exception type that matches this string in the `except` clause. That way your `except` clause is a lot simpler.

For example,

```python
...
except DidNotMatchException:
    logging.debug("go module at %s is downstream only, skip its resetting", module_base_path)
    break

proc = subprocess.run(
...
```

Reply:
See https://github.com/openshift-eng/rebasebot/pull/48#discussion_r1432855978